Parser error when loading JML with nested ternary

## Description

When trying to load a file (Ternary.java) where the following nested ternary expression is used within JML, KeY opens a window titled "Parser Error" with the message "null at .../Ternary.java:2:30  Caused by: java.lang.NullPointerException" and marks `true` as the problem:
```
//@ ensures \result == (((true ? (0 + 1) : 2) > 3) ? 4 : 5);
int foo() {
    return (((true ? (0 + 1) : 2) > 3) ? 4 : 5);
}
```

However, this should be valid Java Syntax – when replacing
`//@ ensures \result == (((true ? (0 + 1) : 2) > 3) ? 4 : 5);`
with
`//@ ensures \result == 5;` in Ternary.java so that now only the Java Code contains the same ternary expression, the file becomes loadable and `foo` provable.

Note: The problem seems to occur only with this specific nesting of ternary operators and a simple expression. It does NOT occur e.g. when omitting `0 + `, when using only one ternary operator or when putting the second ternary operator in place of the `4` (resulting in `((6 > 3) ? (true ? (0 + 1) : 2) : 5)`) or the `5`.

## Reproducible
  
always

### Steps to reproduce 

1. load Ternary.java in KeY
2. experience a Parser Error
3. replace the nested ternary in JML with `5`
4. load Ternary.java again
5. note that the file is now loadable and provable

The file should be loadable and `foo` provable with the use of the nested ternary operator in JML – 2. should not occur.

### Additional information

The "Parser Error" window contains the message
```
The following exception occured:

null at xxx/Ternary.java:2:30

Caused by: java.lang.NullPointerException
```

"Show Details" gives the following stacktrace:

```
null at .../Ternary.java:2:30 (.../Ternary.java:2:30)
	at de.uka.ilkd.key.speclang.njml.Translator.raiseError(Translator.java:2535)
	at de.uka.ilkd.key.speclang.njml.Translator.visitRelational_chain(Translator.java:693)
	at de.uka.ilkd.key.speclang.njml.Translator.visitRelational_chain(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$Relational_chainContext.accept(JmlParser.java:7555)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.oneOf(Translator.java:147)
	at de.uka.ilkd.key.speclang.njml.Translator.visitRelationalexpr(Translator.java:494)
	at de.uka.ilkd.key.speclang.njml.JmlParser$RelationalexprContext.accept(JmlParser.java:7344)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEqualityexpr(Translator.java:562)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEqualityexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$EqualityexprContext.accept(JmlParser.java:7270)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitAndexpr(Translator.java:544)
	at de.uka.ilkd.key.speclang.njml.JmlParser$AndexprContext.accept(JmlParser.java:7201)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExclusiveorexpr(Translator.java:529)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ExclusiveorexprContext.accept(JmlParser.java:7132)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitInclusiveorexpr(Translator.java:514)
	at de.uka.ilkd.key.speclang.njml.JmlParser$InclusiveorexprContext.accept(JmlParser.java:7063)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalandexpr(Translator.java:501)
	at de.uka.ilkd.key.speclang.njml.JmlParser$LogicalandexprContext.accept(JmlParser.java:6994)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalorexpr(Translator.java:482)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalorexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$LogicalorexprContext.accept(JmlParser.java:6925)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitImpliesexpr(Translator.java:446)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ImpliesexprContext.accept(JmlParser.java:6662)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEquivalenceexpr(Translator.java:426)
	at de.uka.ilkd.key.speclang.njml.JmlParser$EquivalenceexprContext.accept(JmlParser.java:6585)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitConditionalexpr(Translator.java:412)
	at de.uka.ilkd.key.speclang.njml.Translator.visitConditionalexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ConditionalexprContext.accept(JmlParser.java:6516)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExpression(Translator.java:402)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExpression(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ExpressionContext.accept(JmlParser.java:6463)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPrimaryParen(Translator.java:1422)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PrimaryParenContext.accept(JmlParser.java:10027)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:46)
	at de.uka.ilkd.key.speclang.njml.JmlParserBaseVisitor.visitPrimaryexpr(JmlParserBaseVisitor.java:724)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PrimaryexprContext.accept(JmlParser.java:8399)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPostfixexpr(Translator.java:854)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPostfixexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PostfixexprContext.accept(JmlParser.java:8315)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitUnaryexprnotplusminus(Translator.java:840)
	at de.uka.ilkd.key.speclang.njml.JmlParser$UnaryexprnotplusminusContext.accept(JmlParser.java:8146)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.oneOf(Translator.java:147)
	at de.uka.ilkd.key.speclang.njml.Translator.visitUnaryexpr(Translator.java:800)
	at de.uka.ilkd.key.speclang.njml.Translator.visitUnaryexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$UnaryexprContext.accept(JmlParser.java:8002)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitMultexpr(Translator.java:737)
	at de.uka.ilkd.key.speclang.njml.JmlParser$MultexprContext.accept(JmlParser.java:7921)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitAdditiveexpr(Translator.java:720)
	at de.uka.ilkd.key.speclang.njml.JmlParser$AdditiveexprContext.accept(JmlParser.java:7829)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitShiftexpr(Translator.java:703)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ShiftexprContext.accept(JmlParser.java:7742)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.oneOf(Translator.java:147)
	at de.uka.ilkd.key.speclang.njml.Translator.visitRelationalexpr(Translator.java:494)
	at de.uka.ilkd.key.speclang.njml.JmlParser$RelationalexprContext.accept(JmlParser.java:7344)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEqualityexpr(Translator.java:562)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEqualityexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$EqualityexprContext.accept(JmlParser.java:7270)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitAndexpr(Translator.java:544)
	at de.uka.ilkd.key.speclang.njml.JmlParser$AndexprContext.accept(JmlParser.java:7201)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExclusiveorexpr(Translator.java:529)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ExclusiveorexprContext.accept(JmlParser.java:7132)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitInclusiveorexpr(Translator.java:514)
	at de.uka.ilkd.key.speclang.njml.JmlParser$InclusiveorexprContext.accept(JmlParser.java:7063)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalandexpr(Translator.java:501)
	at de.uka.ilkd.key.speclang.njml.JmlParser$LogicalandexprContext.accept(JmlParser.java:6994)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalorexpr(Translator.java:482)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalorexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$LogicalorexprContext.accept(JmlParser.java:6925)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitImpliesexpr(Translator.java:446)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ImpliesexprContext.accept(JmlParser.java:6662)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEquivalenceexpr(Translator.java:426)
	at de.uka.ilkd.key.speclang.njml.JmlParser$EquivalenceexprContext.accept(JmlParser.java:6585)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitConditionalexpr(Translator.java:412)
	at de.uka.ilkd.key.speclang.njml.Translator.visitConditionalexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ConditionalexprContext.accept(JmlParser.java:6516)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExpression(Translator.java:402)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExpression(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ExpressionContext.accept(JmlParser.java:6463)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPrimaryParen(Translator.java:1422)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PrimaryParenContext.accept(JmlParser.java:10027)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:46)
	at de.uka.ilkd.key.speclang.njml.JmlParserBaseVisitor.visitPrimaryexpr(JmlParserBaseVisitor.java:724)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PrimaryexprContext.accept(JmlParser.java:8399)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPostfixexpr(Translator.java:854)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPostfixexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PostfixexprContext.accept(JmlParser.java:8315)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitUnaryexprnotplusminus(Translator.java:840)
	at de.uka.ilkd.key.speclang.njml.JmlParser$UnaryexprnotplusminusContext.accept(JmlParser.java:8146)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.oneOf(Translator.java:147)
	at de.uka.ilkd.key.speclang.njml.Translator.visitUnaryexpr(Translator.java:800)
	at de.uka.ilkd.key.speclang.njml.Translator.visitUnaryexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$UnaryexprContext.accept(JmlParser.java:8002)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitMultexpr(Translator.java:737)
	at de.uka.ilkd.key.speclang.njml.JmlParser$MultexprContext.accept(JmlParser.java:7921)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitAdditiveexpr(Translator.java:720)
	at de.uka.ilkd.key.speclang.njml.JmlParser$AdditiveexprContext.accept(JmlParser.java:7829)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitShiftexpr(Translator.java:703)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ShiftexprContext.accept(JmlParser.java:7742)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.oneOf(Translator.java:147)
	at de.uka.ilkd.key.speclang.njml.Translator.visitRelationalexpr(Translator.java:494)
	at de.uka.ilkd.key.speclang.njml.JmlParser$RelationalexprContext.accept(JmlParser.java:7344)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEqualityexpr(Translator.java:562)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEqualityexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$EqualityexprContext.accept(JmlParser.java:7270)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitAndexpr(Translator.java:544)
	at de.uka.ilkd.key.speclang.njml.JmlParser$AndexprContext.accept(JmlParser.java:7201)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExclusiveorexpr(Translator.java:529)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ExclusiveorexprContext.accept(JmlParser.java:7132)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitInclusiveorexpr(Translator.java:514)
	at de.uka.ilkd.key.speclang.njml.JmlParser$InclusiveorexprContext.accept(JmlParser.java:7063)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalandexpr(Translator.java:501)
	at de.uka.ilkd.key.speclang.njml.JmlParser$LogicalandexprContext.accept(JmlParser.java:6994)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalorexpr(Translator.java:482)
	at de.uka.ilkd.key.speclang.njml.Translator.visitLogicalorexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$LogicalorexprContext.accept(JmlParser.java:6925)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitImpliesexpr(Translator.java:446)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ImpliesexprContext.accept(JmlParser.java:6662)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.lambda$mapOf$0(Translator.java:133)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.uka.ilkd.key.speclang.njml.Translator.mapOf(Translator.java:133)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEquivalenceexpr(Translator.java:426)
	at de.uka.ilkd.key.speclang.njml.JmlParser$EquivalenceexprContext.accept(JmlParser.java:6585)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitConditionalexpr(Translator.java:412)
	at de.uka.ilkd.key.speclang.njml.Translator.visitConditionalexpr(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ConditionalexprContext.accept(JmlParser.java:6516)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExpression(Translator.java:402)
	at de.uka.ilkd.key.speclang.njml.Translator.visitExpression(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$ExpressionContext.accept(JmlParser.java:6463)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPredicate(Translator.java:392)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PredicateContext.accept(JmlParser.java:6353)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPredornot(Translator.java:376)
	at de.uka.ilkd.key.speclang.njml.Translator.visitPredornot(Translator.java:63)
	at de.uka.ilkd.key.speclang.njml.JmlParser$PredornotContext.accept(JmlParser.java:6190)
	at de.uka.ilkd.key.speclang.njml.Translator.accept(Translator.java:128)
	at de.uka.ilkd.key.speclang.njml.Translator.visitEnsures_clause(Translator.java:2121)
	at de.uka.ilkd.key.speclang.njml.JmlParser$Ensures_clauseContext.accept(JmlParser.java:2019)
	at de.uka.ilkd.key.speclang.njml.JmlIO.interpret(JmlIO.java:178)
	at de.uka.ilkd.key.speclang.njml.JmlIO.translateTerm(JmlIO.java:188)
	at de.uka.ilkd.key.speclang.njml.JmlIO.translateTerm(JmlIO.java:214)
	at de.uka.ilkd.key.speclang.jml.translation.JMLSpecFactory.translateAndClauses(JMLSpecFactory.java:558)
	at de.uka.ilkd.key.speclang.jml.translation.JMLSpecFactory.translateEnsures(JMLSpecFactory.java:682)
	at de.uka.ilkd.key.speclang.jml.translation.JMLSpecFactory.translateEnsures(JMLSpecFactory.java:436)
	at de.uka.ilkd.key.speclang.jml.translation.JMLSpecFactory.translateJMLClauses(JMLSpecFactory.java:364)
	at de.uka.ilkd.key.speclang.jml.translation.JMLSpecFactory.createJMLOperationContracts(JMLSpecFactory.java:1144)
	at de.uka.ilkd.key.speclang.jml.JMLSpecExtractor.extractMethodSpecs(JMLSpecExtractor.java:476)
	at de.uka.ilkd.key.speclang.SLEnvInput.createSpecs(SLEnvInput.java:314)
	at de.uka.ilkd.key.speclang.SLEnvInput.read(SLEnvInput.java:361)
	at de.uka.ilkd.key.proof.init.ProblemInitializer.readEnvInput(ProblemInitializer.java:334)
	at de.uka.ilkd.key.proof.init.ProblemInitializer.prepare(ProblemInitializer.java:558)
	at de.uka.ilkd.key.proof.init.ProblemInitializer.prepare(ProblemInitializer.java:458)
	at de.uka.ilkd.key.proof.io.AbstractProblemLoader.createInitConfig(AbstractProblemLoader.java:558)
	at de.uka.ilkd.key.proof.io.AbstractProblemLoader.loadEnvironment(AbstractProblemLoader.java:318)
	at de.uka.ilkd.key.proof.io.AbstractProblemLoader.load(AbstractProblemLoader.java:276)
	at de.uka.ilkd.key.proof.io.ProblemLoader.doWork(ProblemLoader.java:64)
	at de.uka.ilkd.key.proof.io.ProblemLoader$1.doInBackground(ProblemLoader.java:117)
	at de.uka.ilkd.key.proof.io.ProblemLoader$1.doInBackground(ProblemLoader.java:110)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException
	at de.uka.ilkd.key.java.TypeConverter.getPromotedType(TypeConverter.java:374)
	at de.uka.ilkd.key.speclang.njml.LDTHandler.build(LDTHandler.java:64)
	at de.uka.ilkd.key.speclang.njml.OverloadedOperatorHandler.build(OverloadedOperatorHandler.java:127)
	at de.uka.ilkd.key.speclang.njml.JmlTermFactory.binary(JmlTermFactory.java:391)
	at de.uka.ilkd.key.speclang.njml.Translator.visitRelational_chain(Translator.java:686)
	... 314 more
```

---

* Commit: 6a929470ef499d0331f85df1bae1a5384116ec49

--- a/key.core/src/main/java/de/uka/ilkd/key/speclang/njml/JmlTermFactory.java
+++ b/key.core/src/main/java/de/uka/ilkd/key/speclang/njml/JmlTermFactory.java
@@ -546,12 +546,13 @@ public final class JmlTermFactory {
 
         Term ife = tb.ife(tb.convertToFormula(result.getTerm()), aTerm, bTerm);
         if (a.getType() != null && b.getType() != null) {
-            if(a.getType().equals(b.getType())) {
+            if (a.getType().equals(b.getType())) {
                 // same type: obvious case
                 result = new SLExpression(ife, a.getType());
             } else {
-                KeYJavaType promotedType = services.getTypeConverter().getPromotedType(a.getType(), b.getType());
-                if(promotedType != null) {
+                KeYJavaType promotedType =
+                    services.getTypeConverter().getPromotedType(a.getType(), b.getType());
+                if (promotedType != null) {
                     // different, put compatible types: add a cast to make sure that
                     // an int is cast to a float e.g.
                     result = new SLExpression(tb.cast(promotedType.getSort(), ife), promotedType);

